ding/policy/decision_transformer.py

```diff
@@ -0,0 +1,409 @@
+        # rtg_target: max target of `return to go`
+        # Our goal is normalize `return to go` to (0, 1), which will favour the covergence. As a result, we usually set rtg_scale == rtg_target.
+        self.rtg_scale = self._cfg.rtg_target  # normalize returns to go
+        self.rtg_target = self._cfg.rtg_target  # max target reward_to_go
+        self.max_eval_ep_len = self._cfg.max_eval_ep_len  # max len of one episode
+        self.num_eval_ep = self._cfg.num_eval_ep  # num of evaluation episodes
+
+        lr = self._cfg.learn.learning_rate  # learning rate
+        wt_decay = self._cfg.wt_decay  # weight decay
+        warmup_steps = self._cfg.warmup_steps  # warmup steps for lr scheduler
+
+        # total updates = max_train_iters x num_updates_per_iter
+        max_train_iters = self._cfg.max_train_iters
+        self.num_updates_per_iter = self._cfg.num_updates_per_iter
+
+        self.context_len = self._cfg.context_len  # K in decision transformer
+        n_blocks = self._cfg.n_blocks  # num of transformer blocks
+        embed_dim = self._cfg.embed_dim  # embedding (hidden) dim of transformer
+        n_heads = self._cfg.n_heads  # num of transformer heads
+        dropout_p = self._cfg.dropout_p  # dropout probability
+
+        # # load data from this file
+        # dataset_path = f'{self._cfg.dataset_dir}/{env_d4rl_name}.pkl'
+
+        # saves model and csv in this directory
+        self.log_dir = self._cfg.log_dir
+        if not os.path.exists(self.log_dir):
+            os.makedirs(self.log_dir)
+
+        # training and evaluation device
+        self.device = torch.device(self._device)
+        device = torch.device(self._device)
+
+        self.start_time = datetime.now().replace(microsecond=0)
+        self.start_time_str = self.start_time.strftime("%y-%m-%d-%H-%M-%S")
+
+        # prefix = "dt_" + env_d4rl_name
+        self.prefix = "dt_" + self.env_name
+
+        save_model_name = self.prefix + "_model_" + self.start_time_str + ".pt"
+        self.save_model_path = os.path.join(self.log_dir, save_model_name)
+        self.save_best_model_path = self.save_model_path[:-3] + "_best.pt"
+
+        log_csv_name = self.prefix + "_log_" + self.start_time_str + ".csv"
+        log_csv_path = os.path.join(self.log_dir, log_csv_name)
+
+        self.csv_writer = csv.writer(open(log_csv_path, 'a', 1))
+        csv_header = (["duration", "num_updates",
+                    "eval_avg_reward", "eval_avg_ep_len", "eval_d4rl_score"])
+
+        self.csv_writer.writerow(csv_header)
+
+        dataset_path = self._cfg.learn.dataset_path
+        print("=" * 60)
+        print("start time: " + self.start_time_str)
+        print("=" * 60)
+
+        print("device set to: " + str(device))
+        print("dataset path: " + dataset_path)
+        print("model save path: " + self.save_model_path)
+        print("log csv save path: " + log_csv_path)
+
+        self._env = gym.make(self.env_name)
```

Comment:
move variable related to eval int `_init_eval` method

Reply:
Actually there is not `_init_eval` function as the difference between dt and other RL methods. We just use `policy.evaluate` function